python/packages/autogen-studio/autogenstudio/web/managers/connection.py

```diff
@@ -97,34 +98,41 @@ async def start_stream(self, run_id: UUID, task: str, team_config: dict) -> None
 
             input_func = self.create_input_func(run_id)
 
+            current_message = ""
             async for message in team_manager.run_stream(tool_context=tool_context,
                 task=task, team_config=team_config, input_func=input_func, cancellation_token=cancellation_token
             ):
                 if cancellation_token.is_cancelled() or run_id in self._closed_connections:
                     logger.info(f"Stream cancelled or connection closed for run {run_id}")
                     break
 
-                formatted_message = self._format_message(message)
-                if formatted_message:
-                    await self._send_message(run_id, formatted_message)
-
-                    # Save messages by concrete type
-                    if isinstance(
-                        message,
-                        (
-                            TextMessage,
-                            MultiModalMessage,
-                            StopMessage,
-                            HandoffMessage,
-                            ToolCallRequestEvent,
-                            ToolCallExecutionEvent,
-                            LLMCallEventMessage,
-                        ),
-                    ):
-                        await self._save_message(run_id, message)
-                    # Capture final result if it's a TeamResult
-                    elif isinstance(message, TeamResult):
-                        final_result = message.model_dump()
+                if isinstance(message, ModelClientStreamingChunkEvent):
+                    current_message += message.content
+                    formatted = self._format_message(message)
+                    if formatted:
+                        await self._send_message(run_id, formatted)
+                else:
+                    formatted_message = self._format_message(message)
+                    if formatted_message:
+                        await self._send_message(run_id, formatted_message)
+
+                        # Save messages by concrete type
+                        if isinstance(
+                            message,
+                            (
+                                TextMessage,
+                                MultiModalMessage,
+                                StopMessage,
+                                HandoffMessage,
+                                ToolCallRequestEvent,
+                                ToolCallExecutionEvent,
+                                LLMCallEventMessage,
+                            ),
+                        ):
+                            await self._save_message(run_id, message)
```

Comment:
n order to resume sessions we're going to figure out how to save these streamed messages as well. We can buffer and save as a `TextMessage` most likely